mimic/rest/auth_api.py

```diff
@@ -2,38 +2,107 @@
 """
 Defines get token, impersonation
 """
-
 import json
-from six import text_type
-from uuid import uuid4
+import attr
 
 from twisted.web.server import Request
 from twisted.python.urlpath import URLPath
-from mimic.canned_responses.auth import get_token, get_endpoints, impersonator_user_role
+from mimic.canned_responses.auth import (
+    get_token,
+    get_endpoints,
+    format_timestamp,
+    impersonator_user_role)
+from mimic.canned_responses.mimic_presets import get_presets
+from mimic.model.behaviors import BehaviorRegistry, EventDescription
+from mimic.model.identity import (
+    APIKeyCredentials,
+    ImpersonationCredentials,
+    PasswordCredentials,
+    TokenCredentials)
 from mimic.rest.mimicapp import MimicApp
-from mimic.canned_responses.auth import format_timestamp
-from mimic.util.helper import invalid_resource
 from mimic.session import NonMatchingTenantError
-from mimic.canned_responses.mimic_presets import get_presets
+from mimic.util.helper import invalid_resource
 
 Request.defaultContentType = 'application/json'
 
 
-class AuthApi(object):
+authentication = EventDescription()
+"""
```

Comment:
I believe that if you want this docstring to document `authentication` it should be in the module docstring as a `:var:`.